header include generated for unused imported variables

### Example

`x.nim`:

```nim
var x* {.importc, header: "does-not-exist".}: int
```

`test.nim`:

```nim
import x
```

### Actual Output

```
$ nim c test.nim

@mx.nim.c:8:10: fatal error: does-not-exist: No such file or directory
```

### Expected Output

Success

### Additional Information

* Found via Windows CI docgen for `posix_utils.nim`
* I'm not sure if this can even be classified as a bug

--- a/compiler/mir/mirgen.nim
+++ b/compiler/mir/mirgen.nim
@@ -1243,10 +1243,10 @@ proc genLocDef(c: var TCtx, n: PNode, val: PNode) =
     s = n.sym
     hasInitializer = val.kind != nkEmpty
     sink = sfCursor notin s.flags
-    node = nameNode(c, s)
+    kind = symbolToPmir(s)
 
   c.builder.useSource(c.sp, n)
-  if node.kind == mnkGlobal and c.scopeDepth == 1:
+  if kind == pirGlobal and c.scopeDepth == 1:
     # no 'def' statement is emitted for top-level globals
     if hasInitializer:
       genAsgn(c, true, sink, n, val)
@@ -1257,7 +1257,7 @@ proc genLocDef(c: var TCtx, n: PNode, val: PNode) =
       # the location doesn't have an explicit starting value. Initialize
       # it to the type's default value.
       c.buildStmt mnkInit:
-        c.add node
+        c.add nameNode(c, s)
         c.buildMagicCall mDefault, s.typ:
           discard
     else:
@@ -1265,7 +1265,7 @@ proc genLocDef(c: var TCtx, n: PNode, val: PNode) =
       discard
   else:
     c.buildStmt (if sfCursor in s.flags: mnkDefCursor else: mnkDef):
-      c.add node
+      c.add nameNode(c, s)
       if hasInitializer:
         genAsgnSource(c, val, sink)
       else:

--- a/compiler/mir/proto_mir.nim
+++ b/compiler/mir/proto_mir.nim
@@ -438,6 +438,25 @@ proc wantArray(e: var seq[ProtoItem]) =
     #      without them, so we do prefer lvalue captures
     e[^1].keep = kLvalue
 
+func symbolToPmir*(s: PSym): range[pirProc..pirConst] =
+  ## Returns the proto-MIR item kind corresponding to `s`.
+  case s.kind
+  of skVar, skLet, skForVar:
+    if sfGlobal in s.flags:
+      pirGlobal
+    else:
+      pirLocal
+  of skTemp, skResult:
+    pirLocal
+  of skParam:
+    pirParam
+  of skConst:
+    pirConst
+  of skProc, skFunc, skConverter, skMethod, skIterator:
+    pirProc
+  else:
+    unreachable(s.kind)
+
 proc exprToPmir(c: TranslateCtx, result: var seq[ProtoItem], n: PNode, sink: bool) =
   ## Translates the single node `n` and recurses if it's a non-terminal. This
   ## procedure makes up the core of the AST-to-proto-MIR translation.
@@ -482,24 +501,7 @@ proc exprToPmir(c: TranslateCtx, result: var seq[ProtoItem], n: PNode, sink: boo
   of nkLambdaKinds:
     node pirProc, sym, n[namePos].sym
   of nkSym:
-    let kind: range[pirProc..pirConst] =
-      case n.sym.kind
-      of skVar, skLet, skForVar:
-        if sfGlobal in n.sym.flags:
-          pirGlobal
-        else:
-          pirLocal
-      of skTemp, skResult:
-        pirLocal
-      of skParam:
-        pirParam
-      of skConst:
-        pirConst
-      of skProc, skFunc, skConverter, skMethod, skIterator:
-        pirProc
-      else:
-        unreachable(n.sym.kind)
-
+    let kind = symbolToPmir(n.sym)
     result.add ProtoItem(orig: n, typ: n.sym.typ, kind: kind, sym: n.sym)
   of nkDerefExpr:
     wantPure(n[0])

--- a/compiler/sem/modulelowering.nim
+++ b/compiler/sem/modulelowering.nim
@@ -190,8 +190,8 @@ proc registerGlobals(stmts: seq[PNode], structs: var ModuleStructs) =
   ## the module level (within the module imperative body `stmts`).
 
   proc register(structs: var ModuleStructs, s: PSym, isTopLevel: bool) {.nimcall.} =
-    if sfCompileTime in s.flags:
-      # don't register compile-time globals with the module struct
+    if {sfCompileTime, sfImportc} * s.flags != {}:
+      # don't register compile-time or imported globals with the module struct
       discard
     elif s.kind == skTemp:
       # HACK: semantic analysis sometimes produces temporaries (it does so for

--- a/tests/lang/s05_pragmas/s01_interop/t05_unused_importc.nim
+++ b/tests/lang/s05_pragmas/s01_interop/t05_unused_importc.nim
@@ -1,0 +1,18 @@
+discard """
+  description: '''
+    Let and var bindings for imported entities are only included in the
+    build if they're assigned to, read from, or have their address taken,
+    within code part of the final build
+  '''
+"""
+
+var a {.importc, header: "<doesnt_exist.h>".}: int
+let b {.importc, header: "<doesnt_exist2.h>".}: int
+
+# an initial assignment (e.g.: `` = 0``) would count as a usage and result in
+# a compiler error, since the headers are non-existent
+
+proc f() =
+  # while this is a usage of `a`, `f` is not part of the final build, so
+  # `a` is not pulled in
+  a = 1

